src/utils/transaction.ts

```diff
@@ -314,6 +524,266 @@ export const buildCrossChainAction = async (
+              symbol: fromAssetSymbol,
+              decimals: fromAssetDecimals,
+              logoURI: fromAssetIconUrl,
+            },
+            toAsset: {
+              address: toAssetAddress,
+              symbol: toAssetSymbol,
+              decimals: toAssetDecimals,
+              logoURI: toAssetIconUrl,
+            },
+            offer,
+            receiverAddress,
+            accountType,
+          },
+        } = transactionBlock;
+
+        const fromAmountBN = ethers.utils.parseUnits(amount, fromAssetDecimals);
+
+        const swapServiceDetails = swapServiceIdToDetails[offer.provider];
+
+        let preview = {
+          fromChainId: chainId,
+          chainId,
+          hasEnoughPLR: transactionBlock?.values?.hasEnoughPLR,
+          fromAsset: {
+            address: fromAssetAddress,
+            decimals: fromAssetDecimals,
+            symbol: fromAssetSymbol,
+            amount: fromAmountBN.toString(),
+            iconUrl: fromAssetIconUrl,
+          },
+          toAsset: {
+            address: toAssetAddress,
+            decimals: toAssetDecimals,
+            symbol: toAssetSymbol,
+            amount: offer.receiveAmount.toString(),
+            iconUrl: toAssetIconUrl,
+          },
+          providerName: swapServiceDetails.title ?? 'Unknown provider',
+          providerIconUrl: swapServiceDetails?.iconUrl,
+          receiverAddress,
+        };
+        let result = await fetchSwapAssetTransaction(
+          chainId,
+          fromAmountBN,
+          fromAssetAddress,
+          offer,
+          receiverAddress,
+          sdk
+        );
+        if (result.errorMessage) return { errorMessage: result.errorMessage };
+
+        const crossChainAction: ICrossChainAction = {
+          id: crossChainActionId,
+          relatedTransactionBlockId: transactionBlock.id,
+          chainId,
+          type: TRANSACTION_BLOCK_TYPE.PLR_DAO_STAKE,
+          preview,
+          transactions: result?.result?.transactions || [],
+          isEstimating: false,
+          estimated: null,
+          destinationCrossChainAction: [],
+          useWeb3Provider: accountType === AccountTypes.Key,
+          receiveAmount: transactionBlock.values.amount,
+          multiCallData: transactionBlock?.multiCallData,
+        };
+
+        return { crossChainAction };
+      } catch (e) {
+        return { errorMessage: 'Failed to build swap transaction!' };
+      }
+    } else {
+      // Staking
+      try {
+        const {
+          values: {
+            fromChainId,
+            fromAsset: {
+              address: fromAssetAddress,
+              symbol: fromAssetSymbol,
+              decimals: fromAssetDecimals,
+              logoURI: fromAssetIconUrl,
+            },
+            amount,
+            offer,
+            accountType,
+          },
+        } = transactionBlock;
+
+        const amountBN = ethers.utils.parseUnits(amount, fromAssetDecimals);
+        try {
```

Comment:
These `try` blocks are pretty large, and ideally we shouldn't be nesting a try inside another try - the first one spans from lines 618-781, and the nested from lines 635-778. 

Could you break these down, and only wrap things that need to be wrapped in try blocks (like interactions with the sdk or a library etc)